src/core/control/Control.cpp

```diff
@@ -2403,6 +2406,62 @@ auto Control::openFile(fs::path filepath, int scrollToPage, bool forceOpen) -> b
     return true;
 }
 
+auto Control::changePdfBackground() -> bool {
+
+    LoadHandler loadHandler;
+
+    const fs::path currentFilepath = ((*this).doc)->getFilepath();
+    const fs::path currentPdfFilepath = ((*this).doc)->getPdfFilepath();
+
+    auto pageNr = getCurrentPageNo();
+    PageRef page = this->doc->getPage(pageNr);
+    // Layer* layer = page->getSelectedLayer();
+
+    if (currentPdfFilepath.empty()) {
```

Comment:
Is this restriction necessary? It might be useful to allow an empty current PDF in case the user wants to attach a PDF background to a document that does not already have one.

Reply:
In principle no, though the logic would be cumbersome I think. §
If you haven't opened a pdf, you can't change the background. For that you should use "Open" or "Annotate PDF". Do you agree?

Reply:
No, I don't see why we would need a limitation like that. For example, suppose I create a blank document but then decide to add some PDF pages later. In this case, it would make sense to allow using the Change PDF Background menu item to append the extra PDF pages.

Reply:
I had a similar discussion (but somewhat in the other direction) in the parent PR https://github.com/xournalpp/xournalpp/pull/4803 with @rolandlo: see e.g. [this comment](https://github.com/xournalpp/xournalpp/pull/4803#pullrequestreview-1561614906).
Could you please suggest a patch on which [you](@Technius) and @rolandlo agree? Thank you in advance.